framework/src/modules/interoperability/sidechain/module.ts

```diff
@@ -195,4 +211,184 @@ export class SidechainInteroperabilityModule extends BaseInteroperabilityModule
 	public async init(_args: ModuleInitArgs) {
 		this._mainchainRegistrationCommand.addDependencies(this._validatorsMethod);
 	}
+
+	// @see https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#mainchain
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async initGenesisState(ctx: GenesisBlockExecuteContext): Promise<void> {
+		const genesisBlockAssetBytes = ctx.assets.getAsset(MODULE_NAME_INTEROPERABILITY);
+		if (!genesisBlockAssetBytes) {
+			return;
+		}
+
+		const genesisInteroperability = codec.decode<GenesisInteroperability>(
+			genesisInteroperabilitySchema,
+			genesisBlockAssetBytes,
+		);
+
+		validator.validate<GenesisInteroperability>(
+			genesisInteroperabilitySchema,
+			genesisInteroperability,
+		);
+
+		await this._verifyChainInfos(ctx, genesisInteroperability);
+	}
+
+	private async _verifyChainInfos(
+		ctx: GenesisBlockExecuteContext,
+		genesisInteroperability: GenesisInteroperability,
+	) {
+		const {
+			ownChainName,
+			ownChainNonce,
+			chainInfos,
+			terminatedStateAccounts,
+			terminatedOutboxAccounts,
+		} = genesisInteroperability;
+
+		// If chainInfos is empty, then check that:
+		//
+		// ownChainName is the empty string;
+		// ownChainNonce == 0;
+		// terminatedStateAccounts is empty;
+		// terminatedOutboxAccounts is empty.
+		if (chainInfos.length === 0) {
+			const ifChainInfosIsEmpty = 'if chainInfos is empty.';
+			if (ownChainName !== '') {
+				throw new Error(`ownChainName must be empty string, ${ifChainInfosIsEmpty}.`);
+			}
+			if (ownChainNonce !== BigInt(0)) {
+				throw new Error(`ownChainNonce must be 0, ${ifChainInfosIsEmpty}.`);
+			}
+			if (terminatedStateAccounts.length !== 0) {
+				throw new Error(`terminatedStateAccounts must be empty, ${ifChainInfosIsEmpty}.`);
+			}
+			if (terminatedOutboxAccounts.length !== 0) {
+				throw new Error(`terminatedOutboxAccounts must be empty, ${ifChainInfosIsEmpty}.`);
+			}
+		} else {
+			// ownChainName
+			// has length between MIN_CHAIN_NAME_LENGTH and MAX_CHAIN_NAME_LENGTH,
+			// is from the character set a-z0-9!@$&_.,
+			// and ownChainName != CHAIN_NAME_MAINCHAIN;
+			if (
+				ownChainName.length < MIN_CHAIN_NAME_LENGTH ||
+				ownChainName.length > MAX_CHAIN_NAME_LENGTH // will only run if not already applied in schema
+			) {
+				throw new Error(
+					`ownChainName.length must be between ${MIN_CHAIN_NAME_LENGTH} and ${MAX_CHAIN_NAME_LENGTH}`,
+				);
+			}
+			// CAUTION!
+			// this check is intentionally applied after MIN_CHAIN_NAME_LENGTH, as it will fail for empty string
+			if (!isValidName(ownChainName)) {
+				throw new Error(`ownChainName must have only ${validNameCharset} character set.`);
+			}
+			if (ownChainName === CHAIN_NAME_MAINCHAIN) {
+				throw new Error(`ownChainName must be not equal to ${CHAIN_NAME_MAINCHAIN}.`);
+			}
+
+			// ownChainNonce > 0
+			if (ownChainNonce < 1) {
```

Comment:
Optional suggestion to be consistent with mainchain code:
```suggestion
			if (ownChainNonce <= 0) {
```